src/collection/bundle-file-collection.js

```diff
@@ -0,0 +1,38 @@
+var _ = require('underscore'),
+    path = require('path'),
+    styleguide = require('../styleguide'),
+    StyleguideBundleError = require('./styleguide-bundle-error.js');
+
+var BundleFileCollection = function(bundleFile) {
+    this._bundleFile = path.resolve(bundleFile);
+    this._isStyleguideBundle = styleguide.isStyleguideFile(bundleFile);
+    this._files = [];
+};
+
+var isAbsolutePath = function(filePath) {
+    return path.resolve(filePath) === path.normalize(filePath);
+};
+
+var resolvePath = function(bundleFile, filePath) {
+    if (isAbsolutePath(filePath)) {
+        return filePath;
+    }
+
+    return path.resolve(path.dirname(bundleFile), filePath);
+};
+
+BundleFileCollection.prototype.addFile = function(filePath) {
+    var absolutePath = resolvePath(this._bundleFile, filePath);
+
+    if (!this._isStyleguideBundle && styleguide.isStyleguideFile(absolutePath)) {
+        throw new StyleguideBundleError(this._bundleFile, absolutePath);

```

Comment:
This is where the check happens. If we're working with a non-styleguide bundle, we throw an error when you try to add a styleguide file to the bundle.